src/app/routes/registration/enrolment-form/enrolment-form.component.ts

```diff
@@ -66,20 +68,31 @@ export class EnrolmentFormComponent implements EnrolmentForm {
   @Output() submitForm = new EventEmitter<EnrolmentSubmission>();
 
   private fields: IFieldDefinition[];
-  isValidSchema = true;
+  isValidSchema: Record<string, boolean> = {};
```

Comment:
In `getControl` (copied below for reference), the controls are indexed by `id`. Do you that would work here (instead of indexing by `label`)? I think that might be a bit better as I think there could, in principle, be duplicate label which might break this implementation.
```ts
getControl(id: number): FormControl {
    return (this.enrolmentForm?.get('fields') as FormArray)?.controls[
      id
    ] as FormControl;
  }
```

Reply:
it might work, but `id` is generated by angular ngFor. So it is safer to not use that for checks like this.